[Bug][Doc]Update database init instruction docs (#9598)

EricGao888 opened the pull request:
## Purpose of the pull request
* Update database initialization instructions in docs
* This PR closes: #9598 

## Brief change log
1. Add instructions on how to create db, user and grant permissions for dolphin users who use latest versions of mysql.
2. Update instructions on how to init db since related directory structure has changed.

## Verify this pull request

* Verified by manual test.

EricGao888 committed:
[Bug][Doc]Update database init instruction docs (#9598)

SbloodyS commented on docs/docs/en/guide/installation/pseudo-cluster.md:
````diff
@@ -146,10 +146,29 @@ mysql> GRANT ALL PRIVILEGES ON dolphinscheduler.* TO '{user}'@'localhost' IDENTI
 mysql> flush privileges;
 ```
 
+The script above does not work for latest verisons (e.g. 8.0.27) of mysql, use the script below instead:
````
I think we should not use this description. Just put ```Mysql 8``` in this area. And ```Mysql 5.6/5.7``` above.

EricGao888 replied:
Sure, I will fix it.

zhongjiajie reviewed (dismissed):
LGTM, does @SbloodyS have addition advice?

SbloodyS commented:
May be we should modifiy this docs too. Because init script and upgrade script are similar. @EricGao888 

https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/guide/upgrade.md

EricGao888 commented:
> May be we should modifiy this docs too. Because init script and upgrade script are similar. @EricGao888
> 
> https://github.com/apache/dolphinscheduler/blob/dev/docs/docs/en/guide/upgrade.md

Sure, I will update it.

zhongjiajie commented:
I find out keyword `create-dolphinscheduler.sh` also in https://raw.githubusercontent.com/apache/dolphinscheduler/9f84dbbda00956d5bf198ce7eb4c47f2ea335173/docs/docs/en/architecture/configuration.md . and the binary package structure already changed, I think you could change it too. But you could also just delete `<create|upgarde>-dolphinscheduler.sh` in this PR too

EricGao888 commented:
> I find out keyword `create-dolphinscheduler.sh` also in https://raw.githubusercontent.com/apache/dolphinscheduler/9f84dbbda00956d5bf198ce7eb4c47f2ea335173/docs/docs/en/architecture/configuration.md . and the binary package structure already changed, I think you could change it too. But you could also just delete `<create|upgarde>-dolphinscheduler.sh` in this PR too

@zhongjiajie Sure. I just figured out there's also a bug in upgrade-schema.sh caused by incorrect file path. I will also fix it in this PR. https://github.com/apache/dolphinscheduler/blob/1de7f154a38566f4d7989c8c7ebe8bda16a8dc6e/dolphinscheduler-tools/src/main/bin/upgrade-schema.sh#L22-L27

EricGao888 committed:
[Bug][Tools]Fix bug in upgrade-schema.sh (#9598)

EricGao888 committed:
[Bug][Doc]Update db upgrade docs (#9598)

EricGao888 commented:
> I find out keyword `create-dolphinscheduler.sh` also in https://raw.githubusercontent.com/apache/dolphinscheduler/9f84dbbda00956d5bf198ce7eb4c47f2ea335173/docs/docs/en/architecture/configuration.md . and the binary package structure already changed, I think you could change it too. But you could also just delete `<create|upgarde>-dolphinscheduler.sh` in this PR too

@SbloodyS @zhongjiajie  I've fixed `upgrade-schema.sh` and updated db upgrade related docs in the latest commits. I just checked the `configuration.md`, it seems that we need to update the whole `Directory Structure` section. I will submit a new PR to this issue: #9670

SbloodyS commented on docs/docs/zh/guide/upgrade.md:
```diff
@@ -13,24 +13,17 @@
-      spring.datasource.url=jdbc:mysql://xxx:3306/dolphinscheduler?useUnicode=true&characterEncoding=UTF-8&allowMultiQueries=true     需要修改ip，本机localhost即可
-      spring.datasource.username=xxx						需要修改为上面的{user}值
-      spring.datasource.password=xxx						需要修改为上面的{password}值
+- 将`./tools/conf/application.yaml`中的username和password改成你设定数据库用户名和密码
```
We should also modify the english version.

EricGao888 replied:
I think I've already updated the English version? 
![image](https://user-images.githubusercontent.com/34905992/164687186-2d72a12d-feb6-4c38-8bac-493e1be283e2.png)

SbloodyS replied:
Sorry. I missed it. @EricGao888

SbloodyS reviewed (approved):
LGTM. Could you take a look when you have spare time? @zhongjiajie

zhongjiajie reviewed (approved):
lgtm, thanks
